java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectHistoryEntry.java

```diff
@@ -168,7 +180,7 @@ public String toString() {
         return new ToStringBuilder(this)
                 .append("id", id)
                 .append("version", version)
-                .append("contentProject", contentProject.getLabel())
+                .append("contentProject", ofNullable(contentProject).map(p -> p.getLabel()).orElse("null"))
```

Comment:
Maybe the usage of `map` here is more a workaround as per `contentProject` is not exactly a `list`?

What about
`.append("contentProject", Opt.fold(Optional.ofNullable(contentProject), () -> "null", p -> p.getLabel()))`
or
`.append("contentProject", ofNullable(contentProject).isPresent() ? contentProject.getLabel() : "null")`

Reply:
I find the alternatives far less readable than the original. It needed to read them couple of times till I got what's going on. Maybe the plain old java < 8 way would be the best:

```java
contentProject == null ? null : contentProject.getLabel()
```

Reply:
> I find the alternatives far less readable than the original..

..functional developers will get :angry: about this - :rofl:

Fine by me using the old plain java, I just thought the usage of `map` was not the logic solution, any replacement is fine :+1: